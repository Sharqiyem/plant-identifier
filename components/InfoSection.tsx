import React from 'react';
import { View, Text } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import Colors from '@/constants/Colors';

export const InfoSection: React.FC = () => (
  <View className="space-x-2 gap-4 bg-card items-center">
    <Ionicons name="information-circle-outline" color={Colors.primary} size={32} />

    <View className=" mb-2">
      <Text className="text-center text-lg text-foreground">
        To start identifying plants, select an image from your gallery or take a photo using the
        camera. Our AI will analyze the image and provide you with detailed information about the
        plant.
      </Text>
    </View>
  </View>
);
