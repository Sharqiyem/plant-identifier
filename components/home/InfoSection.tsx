import React from 'react';
import { View, Text } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import Colors from '@/constants/Colors';

export const InfoSection: React.FC = () => (
  <View className="p-4 bg-card rounded-lg shadow-md">
    <View className="flex-row items-center justify-center mb-3">
      <Ionicons name="information-circle-outline" color={Colors.primary} size={24} />
      <Text className="ml-2 text-lg font-semibold text-primary">How to Use</Text>
    </View>
    <Text className="text-center text-base text-foreground leading-6">
      To identify plants, select an image from your gallery or take a photo using the camera. Our AI
      will analyze the image and provide detailed information about the plant.
    </Text>
  </View>
);
