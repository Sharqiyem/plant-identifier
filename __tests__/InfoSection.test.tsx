import { InfoSection } from '@/components/home/InfoSection';
import { render } from '@testing-library/react-native';
import * as React from 'react';

describe('InfoSection', () => {
  it(`renders correctly`, () => {
    const tree = render(<InfoSection />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`renders match with snapshot`, () => {
    const { getByText } = render(<InfoSection />);
    expect(getByText('How to Use')).toBeDefined();
  });
});
