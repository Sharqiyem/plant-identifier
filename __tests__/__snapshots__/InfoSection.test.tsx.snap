// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoSection renders correctly 1`] = `
<View
  style={
    [
      {
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
      {
        "backgroundColor": "hsl(221.2, 50%, 10%)",
      },
      {
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
      },
      {
        "shadowColor": "rgba(0, 0, 0, 0.1)",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 10,
      },
      {
        "shadowColor": "rgba(0, 0, 0, 0.1)",
        "shadowOffset": {
          "height": 6,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 10,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
        {
          "alignItems": "center",
        },
        {
          "justifyContent": "center",
        },
        {
          "marginBottom": 12,
        },
      ]
    }
  >
    <Text />
    <Text
      style={
        [
          {
            "marginLeft": 8,
          },
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
          {
            "fontWeight": "600",
          },
          {
            "color": "hsl(221.2, 83.2%, 53.3%)",
          },
        ]
      }
    >
      How to Use
    </Text>
  </View>
  <Text
    style={
      [
        {
          "textAlign": "center",
        },
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        {
          "color": "hsl(221.2, 5%, 90%)",
        },
        {
          "lineHeight": 24,
        },
      ]
    }
  >
    To identify plants, select an image from your gallery or take a photo using the camera. Our AI will analyze the image and provide detailed information about the plant.
  </Text>
</View>
`;
